src/openzaak/components/documenten/api/filters.py

```diff
@@ -30,13 +33,119 @@
 logger = logging.getLogger(__name__)
 
 
+class ZaakFilter(FkOrUrlFieldFilter):
+    def filter(self, qs, value):
+        if not value:
+            return qs
+
+        if not check_path(value, "zaken"):
+            logger.debug(
+                "URL (%s) is not a zaak, filtering to empty result set.", value
+            )
+            return qs.none()
+
+        return super().filter(qs, value)
+
+
 class EnkelvoudigInformatieObjectListFilter(FilterSet):
-    trefwoorden = CharArrayFilter(field_name="trefwoorden", lookup_expr="contains")
+    creatiedatum__gte = filters.DateFilter(
+        help_text=mark_experimental(
+            "De aanmakings datum van dit informatie object (groter of gelijk aan de gegeven datum)."
+        ),
+        field_name="creatiedatum",
+        lookup_expr="gte",
+    )
+    creatiedatum__lte = filters.DateFilter(
+        help_text=mark_experimental(
+            "De aanmakings datum van dit informatie object (kleiner of gelijk aan de gegeven datum)."
+        ),
+        field_name="creatiedatum",
+        lookup_expr="lte",
+    )
+    auteur = filters.CharFilter(
+        help_text=mark_experimental(
+            "De persoon of organisatie die dit informatie object heeft aangemaakt."
+        ),
+        lookup_expr="icontains",
+    )
+    beschrijving = filters.CharFilter(
+        help_text=mark_experimental("De bescrijving van dit informatie object."),
+        lookup_expr="icontains",
+    )
+    locked = filters.BooleanFilter(
+        help_text=mark_experimental(
+            "De indicatie of dit informatie object gelocked is of niet."
+        ),
+        method="locked_filter",
+    )
+    titel = filters.CharFilter(
+        help_text=mark_experimental("De titel van het informatie object."),
+        lookup_expr="icontains",
+    )
+    trefwoorden = CharArrayFilter(
+        help_text=mark_experimental(
+            "Filter op de trefwoorden die eventueel gelinkt zijn aan het informatie object."
+        ),
+        lookup_expr="overlap",
```

Comment:
Why has the lookup been changed? This is a deviation of the standard. In the OAS there is no mentioning that it can be `OR` and in the reference implementation they also don't use `overlap`.
I think we should leave it as it was

Reply:
I wasn't aware the `trefwoorden` filter was already part of the OAS. Let's keep the default behaviour in that case. We probably need some way to differentiate between AND and ORs...later :)